lib/api/authentication/web_authentication.py

```diff
@@ -0,0 +1,57 @@
+from django.conf import settings
+from django.middleware.csrf import CsrfViewMiddleware
+import jwt
+
+class WebToken:
+    def __init__(self, user_id, course):
+        self.user_id = user_id
+        self.course = course
+
+    def verify_user(self, user):
+        return self.user_id == user.id
+
+class CSRFCheck(CsrfViewMiddleware):
+    def _reject(self, request, reason):
+        # Return the failure reason instead of an HttpResponse
+        return reason
+
+class SessionAuthentication:
+    """
+    Use Django's session framework for authentication.
+    """
+
+    def authenticate(self, request):
+        """
+        Returns a `User` if the request session currently has a logged in user.
+        Otherwise returns `None`.
+        """
+        # Get the session-based user from the underlying HttpRequest object
+        user = getattr(request._request, 'user', None)
+
+        # Unauthenticated, CSRF validation not required
+        if not user or not user.is_active:
+            return None
+
+        self.enforce_csrf(request)
+        
+        if 'Authorization' in request.headers:
+            token = request.headers['Authorization'].split()[1]
```

Comment:
Remember to check that the first word is `Bearer`.

Reply:
Is it ok to handle somehow faulty tokens as no token, eg missing bearer or invalid jwt, if not what would be better?

Reply:
And especially that there are two parts after splitting, otherwise the server simply crash when submitting a request with nice headers.